### PR TITLE
Add CodeQL Security Scan

### PR DESCRIPTION
**Pull request by xukaren**
## Motivation
Follow up to issue open-telemetry/oteps#144

[CodeQL](https://github.com/github/codeql) is GitHub's static analysis engine which scans repos for security vulnerabilities. As the project grows and we near GA it might be useful to have a workflow which checks for security vulnerabilities with every PR so we can ensure every incremental change is following best development practices. Also passing basic security checks will also make sure that there aren't any glaring issues for our users.

## Changes
This PR adds CodeQL security checks to the repo
* After every run the workflow uploads the results to GitHub. Details on the run and security alerts will show up in the security tab of this repo.

**Workflow Triggers**
* daily cron job at 1:30am
* workflow_dispatch (in case maintainers want to trigger a security check manually)

cc- @alolita

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ All notable changes to this project will be documented in this file.
 * [#54](https://github.com/open-telemetry/opentelemetry-js-api/pull/54) feat: add tracer.startActiveSpan() ([@naseemkullah](https://github.com/naseemkullah))
 * [#58](https://github.com/open-telemetry/opentelemetry-js-api/pull/58) chore: use spancontext for link ([@dyladan](https://github.com/dyladan))
 * [#51](https://github.com/open-telemetry/opentelemetry-js-api/pull/51) feat: add function to wrap SpanContext in NonRecordingSpan #49 ([@dyladan](https://github.com/dyladan))
+* [#75](https://github.com/open-telemetry/opentelemetry-js-api/pull/75) ci: add codeQL security scan ([@xukaren](https://github.com/xukaren))
```
Comment (dyladan) [inline CHANGELOG.md:24]: changelog is automatically generated in this repo. No need to manually bump it

**Review by dyladan — COMMENTED** [review]
Looks good other than the modified changelog

File: CHANGELOG.md (hunk 1 above)
Comment (xukaren) [reply CHANGELOG.md]: Thanks, updated.

**Review by vmarchaud — APPROVED** [review]

**Review by naseemkullah — APPROVED** [review]

**Comment by naseemkullah** [comment]
lgtm 🔐

**Review by johnbley — APPROVED** [review]

**Review by obecny — APPROVED** [review]
lgtm

**Review by dyladan — APPROVED** [review]

**Comment by dyladan** [comment]
Out of date branch and maintainers don't have permission to write to the source branch so I can't merge this until it is updated.

**Comment by xukaren** [comment]
Rebased!

**Comment by vmarchaud** [comment]
@xukaren Sorry but you will need to rebase again :/ Could you allow us to write on your fork so we don't need to bother you in the future ?

**Comment by vmarchaud** [comment]
@xukaren Thanks !

**Pull request merged**